modules/mesh-layers/src/scenegraph-layer/scenegraph-layer.js

```diff
@@ -241,8 +241,8 @@ export default class ScenegraphLayer extends Layer {
   draw({moduleParameters = null, parameters = {}, context}) {
     if (!this.state.scenegraph) return;
 
-    if (this.props._animations && this.state.animator) {
-      this.state.animator.animate(context.animationProps.time);
+    if (this.props._animations && this.state.animator && context.deck) {
+      this.state.animator.animate(context.deck.animationLoop.timeline.time);
```

Comment:
`context.timeline.getTime()`

Reply:
My old code is fixed when using `deck.animationLoop.timeline`, and `context.timeline` didn't fix it.

Do you know the differences, and how they should be used? I don't know how to play `context.timeline` (time doesn't move forward when I used it in my app). But `deck.animationLoop.timeline` is playable through the deck object.

Reply:
They should be the same object : https://github.com/visgl/deck.gl/blob/master/modules/core/src/lib/deck.js#L586

Reply:
Interesting, they don't behave like they are. Where does the context object get set? This code looks like its setting `deck.animationLoop.timeline` and `context.timeline` is set up somewhere else.

Reply:
https://github.com/visgl/deck.gl/blob/7b3b615edb86c7a6adf9230bdcb139a58e9933d3/modules/core/src/lib/deck.js#L584-L586

I just tested in layer-browser and they are the same object.

Reply:
Created an isolated example at https://github.com/visgl/deck.gl/tree/chr-animated-gltf/examples/website/animated-scenegraph to test out the two timeline object references. It seems like the intention the code is for these two objects to be the same, but I wonder if somewhere they get memoized, redefined, or copied such that they don't reference the same object by the time scenegraph layer is executed. 

Where does a layer's `context` get defined?

Here's what version with `context.deck.animationLoop.timeline.getTime()` looks like.
![dce142b9-384d-ec9c-73e4-86f7d5088eed](https://user-images.githubusercontent.com/2461547/96078188-807c1280-0e66-11eb-94fe-8eb52e5281d6.gif)

And this with `context.timeline.getTime()`
![f5474eff-3e9e-273f-a918-cdaa580dde0c](https://user-images.githubusercontent.com/2461547/96078201-883bb700-0e66-11eb-98da-ee898a62797b.gif)

I'm using hubble.gl to record these, which is using `deck.animationLoop.timeline.play()` on a reference of the deck object.

Reply:
Are the object passed to `LayerManager` a layer's `context` eventually?

If so I can see how we expect the timeline nested in deck and the timeline at the surface should be the same.